src/modules/UI/scenes/AddToken/AddToken.ui.js

```diff
@@ -20,7 +20,12 @@ export type AddTokenOwnProps = {
   addNewToken: Function,
   currentCustomTokens: Array<CustomTokenInfo>,
   wallet: GuiWallet,
-  onAddToken: string => void
+  onAddToken: Function,
+  // adding properties in case coming from Scan scene (scan QR code to add token)
+  currencyName?: string,
+  currencyCode?: string,
+  contractAddress?: string,
+  decimalPlaces?: string
```

Comment:
If these are being set in the default state as = to a prop or an empty string
`currencyName: this.props.currencyName || ''`
then we do not need to make these optional.

Reply:
Good point.